### DISCUSSION TITLE
Is there an API that can provide count of active nodes in a cluster?

### DISCUSSION THREAD

**Discussion by dipaktank** [rabbitmq/rabbitmq-server#5786] [Questions] [answered]
Hi,

We wanted to know whether any API available that can provide count of active nodes not configured node in a cluster?

Ex. Let's say I've set up a cluster with two nodes. Now I've down one node, the API should return result 1.

Thanks

**Comment by michaelklishin** [comment, marked as answer]
Clusters of two nodes are [explicitly not supported](https://www.rabbitmq.com/clustering.html#node-count).

There are node metrics, including partitions in effect, exposed by the HTTP API.
With Prometheus all metrics are node-local, so the list of nodes becomes the list of nodes that report metrics. IIRC partitions in effect is still one of the exposed metrics.

The definition of "down" in distributed systems is broad, so other metrics can be of high relevance as well.

**Comment by michaelklishin** [comment]
`GET /api/nodes` is the HTTP API endpoint that would be a good start. With Prometheus, all metrics are reported for a single node, so it's implied that you have an external way of tracking expected cluster members.

**Reply by dipakTankTR** [reply]
@michaelklishin  Thanks for the reply. I will check it.

**Reply by dipakTankTR** [reply]
@michaelklishin Do you have idea How I can produce RabbitMQ network partition in Windows?

**Reply by michaelklishin** [reply]
Same as everywhere, using a specialized proxy such as https://github.com/Shopify/toxiproxy or a traffic shaper, or simply disable the network interface one way or another.
